zebra-network/src/peer_set/inventory_registry.rs

```diff
@@ -125,64 +118,27 @@ impl InventoryChange {
     }
 }
 
-impl<T: Clone> InventoryStatus<T> {
-    /// Returns true if the inventory item was advertised.
-    #[allow(dead_code)]
-    pub fn is_advertised(&self) -> bool {
-        matches!(self, Advertised(_))
-    }
-
-    /// Returns true if the inventory item was missing.
-    #[allow(dead_code)]
-    pub fn is_missing(&self) -> bool {
-        matches!(self, Missing(_))
-    }
-
-    /// Get the advertised inventory item, if present.
-    pub fn advertised(&self) -> Option<T> {
-        if let Advertised(item) = self {
-            Some(item.clone())
-        } else {
-            None
-        }
-    }
-
-    /// Get the rejected inventory item, if present.
-    #[allow(dead_code)]
-    pub fn missing(&self) -> Option<T> {
-        if let Missing(item) = self {
-            Some(item.clone())
-        } else {
-            None
-        }
-    }
-
-    /// Get the inner item, regardless of status.
-    pub fn inner(&self) -> T {
-        match self {
-            Advertised(item) | Missing(item) => item.clone(),
-        }
-    }
-
+impl<T> InventoryStatus<T> {
     /// Get a marker for the status, without any associated data.
     pub fn marker(&self) -> InventoryMarker {
         self.as_ref().map(|_inner| ())
     }
 
     /// Maps an `InventoryStatus<T>` to `InventoryStatus<U>` by applying a function to a contained value.
-    pub fn map<U: Clone, F: FnOnce(T) -> U>(self, f: F) -> InventoryStatus<U> {
-        // Based on Option::map from https://doc.rust-lang.org/src/core/option.rs.html#829
+    pub fn map<U, F: FnOnce(T) -> U>(self, f: F) -> InventoryStatus<U> {
+        // Based on Option::map from https://doc.rust-lang.org/src/core/option.rs.html#844
         match self {
-            Advertised(item) => Advertised(f(item)),
+            Available(item) => Available(f(item)),
             Missing(item) => Missing(f(item)),
         }
     }
+}
 
-    /// Converts from `&InventoryStatus<T>` to `InventoryStatus<&T>`.
-    pub fn as_ref(&self) -> InventoryStatus<&T> {
+impl<T: Clone> InventoryStatus<T> {
+    /// Get the inner item, regardless of status.
+    pub fn inner(&self) -> T {
```

Comment:
Optional: I'd suggest renaming this `to_inner`, so that it gives a hint that the type is being cloned (inspired by the [Rust API guidelines](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv)).

```suggestion
    pub fn to_inner(&self) -> T {
```

Reply:
Fixed in 7c32990c3 in PR #3505